mpg321: add page (#4082)
tldr-pages/tldr: pages/common/mpg321.md

# mpg321

> High Performance MPEG 1.0/2.0/2.5 Audio Player for Layer 1, 2, and 3.
> Mpg321 was written (sometime in 1999) to be a drop-in replacement for the (previously) non-free mpg123 player.
> More information: <http://mpg321.sourceforge.net/>.

- Play an audio source exactly N times (N=0 means forever):

`mpg321 -l {{N}} {{path/to/file_a|URL}} {{path/to/file_b|URL}} {{...}}`

> - Play a whole directory (recursively):

Reviewer: Is there a need to keep "recursively" in parentheses ? The sentence works fine without it.

Author: removed the parentheses.

`mpg321 -B {{path/to/directory}}`

- Enable Basic Keys ( `*` or `/` - Increase or decrease volume, `n` - Skip song, `m` - Mute/unmute.) while playing:

`mpg321 -K {{path/to/file_a|URL}} {{path/to/file_b|URL}} {{...}}`

- Play files randomly until interrupted:

`mpg321 -Z {{path/to/file_a|URL}} {{path/to/file_b|URL}} {{...}}`

- Shuffle list of files before playing the list once:

`mpg321 -z {{path/to/file_a|URL}} {{path/to/file_b|URL}} {{...}}`

> - Play all the files in the current folder(s) and subfolders, randomly (until interrupted), with *Basic Keys* enabled:

Reviewer: Since we are keeping this, please remove the asterisks from "Basic Keys".

Author: Done. Sorry about it - I didn't see the asterisks there...

> `mpg321 -B -Z -K *.mp3`

Reviewer: `*.mp3` within tokens (`{{}}`).

Author: replaced with '.' and updated the text

# mpg321

> > High Performance MPEG 1.0/2.0/2.5 Audio Player for Layer 1, 2, and 3.

Reviewer 2: Perhaps this will read better?
```suggestion
> High Performance MPEG 1.0/2.0/2.5 Audio Player for layers 1, 2, and 3.
```

Author: No. The standard expression for MP3&Co is "Layer 1,2,3" - for example: "To play MP3 audio with XMMS, you need to add the MPEG Layer 1/2/3 Player plug-in."

> > Mpg321 was written to be a drop-in replacement for the (by-then) non-free mpg123 player.

Reviewer 2: Perhaps this will read better?
```suggestion
> Drop-in replacement for the (previously) non-free `mpg123`.
```

Author: Nope, I'd keep it (and even add some more historic info).

> More information: <http://mpg321.sourceforge.net/>.

> - Play an audio source (on repeat) exactly N=3 times:

Reviewer 2: Perhaps this will read better?
```suggestion
- Play an audio source exactly 3 times:
```

> `mpg321 -l {{3}} {{ file(s) | URL }}`

Reviewer 2: Perhaps this will read better?
```suggestion
`mpg321 -l {{3}} {{path/to/file_a|URL}} {{path/to/file_b|URL}} {{...}}`
```

> - Play an audio source (on repeat) forever (0 means until interrupted):

Reviewer 2: Perhaps this will read better?
```suggestion
- Play an audio source on repeat forever (0 means until interrupted):
```

> `mpg321 -l 0 {{ file(s) | URL }}`

Reviewer 2: Same as above.
```suggestion
`mpg321 -l 0 {{path/to/file_a|URL}} {{path/to/file_b|URL}} {{...}}`
```

> `mpg321 -l 0 {{path/to/file_a|URL}} {{path/to/file_b|URL}} {{...}}`

Reviewer: I think we can combine these 2 examples into one. We can just mention `-l {{3}}` and write the description as - "Play an audio source 3 times (0 means forever):"

Author: ok. Replaces the 2 lines with: 'Play an audio source exactly N times (N=0 means forever)'

- Play a whole folder (recursively):

> `mpg321 -B {{ folder(s) }}`

Reviewer 2: Perhaps this will read better?
```suggestion
`mpg321 -B {{path/to/directory}}`
```

> - Play a whole folder (recursively):

Reviewer: "whole directory" to be consistent.

Author: ok, done!

> - Enable Basic Keys ( * or / - Increase or decrease volume, n - Skip song, m - Mute/unmute.) while playing:

Reviewer 2: Perhaps this will read better?
```suggestion
- Enable basic keys (`*` or `/`: increase or decrease volume, `n`: Skip song, `m`: Mute / unmute) while playing:
```

Author: OK, though Basic Keys is the term used in the documentation...

> `mpg321 -K {{ file(s) | URL }}`

Reviewer 2: Same as above.
```suggestion
`mpg321 -K {{path/to/file_a|URL}} {{path/to/file_b|URL}} {{...}}`
```

> - Enable *Basic Keys* ( `*` or `/` - Increase or decrease volume, `n` - Skip song, `m` - Mute/unmute.) while playing:

Reviewer: Markdown in descriptions isn't really supported apart from backticks. I would suggest to remove the asterisks.

Author: ok, done!

- Play files randomly until interrupted:

> `mpg321 -Z {{ file(s) | URL }}`

Reviewer 2: Same as above.
```suggestion
`mpg321 -Z {{path/to/file_a|URL}} {{path/to/file_b|URL}} {{...}}`
```

- Shuffle list of files before playing the list once:

> `mpg321 -z {{ file(s) | URL }}`

Reviewer 2: Perhaps this will read better?
```suggestion
`mpg321 -z {{path/to/file_a|URL}} {{path/to/file_b|URL}} {{...}}`
```

> - Play all the files in the current folder(s) and subfolders, randomly (until interrupted), with Basic keys enabled:

Reviewer 2: This example just seems to combine all the other examples, and doesn't show anything new. To this end, I'm not sure we need to include it. Perhaps we could remove it?

Author: I'd rather leave it, because it shows a quick way to use multiple flags/functionalities.

Author: I, for one, use it often, btw...

> - Play all the files in the current folder(s) and subfolders, randomly (until interrupted), with *Basic Keys* enabled:

Reviewer: We usually leave the combination of multiple options to the users themselves. It is expected that a command will work by combining the individual flags. Wondering if it still makes sense to have this?

Author: Sincerely, yes.... actually this is MY main reason for creating this tldr page - I didn't remember the flags, and I always had to look into its man docs for the flags.... and this is what I expect from tldr: to give me - useful examples.
Besides, what would be the gain in removing this examples?! that we'd have one line less in this file? I strongly suggest to leave this example as it is!

Reviewer: From what I could understand, the flags are already covered in the previous examples. We just need to combine them to get the functionality desired.

My reasoning was to remain consistent with the rest of the pages in the repo. Because this is what we usually follow and encourage our users to build up combos themselves. But it's fine if you want to keep it. Thanks.

Author: Thank you, too!
